web/src/Tests/AppRouter.test.js

```diff
@@ -0,0 +1,30 @@
+import { fireEvent, render, screen} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import App from "../App";
+import { BrowserRouter, MemoryRouter } from "react-router-dom";
+
+//Testing Home page route -> Renders HomePage.js component
+test("Home Page must be the first Page to be rendered", async () => {
+  render(<App />, { wrapper: BrowserRouter });
+  expect(screen.getByText("Homepage")).toBeInTheDocument();
+});
+
+//Testing Topic Page route -> Renders TopicPage.js component
+test("navigating from Home page to Topic Page", async () => {
+  render(<App />, { wrapper: BrowserRouter });
+  const topic = "javascript";
+  fireEvent.click(screen.getByText(topic));
```

Comment:
```suggestion
const topicLink = screen.getAllByRole('link') //find any link
const topicName = topicLink.textContent
fireEvent.click(topicLink)
expect(screen.getByText(topicName)).toBeInTheDocument();
```
In the event that the Javascript link doesn't exist, it may be better for longevity to just find any link, grab its text content and then make sure that text content is at that specific endpoint. You might need to play around with RegExp in order to make sure you don't have case-sensitivity issues.

Reply:
that sounds like a better way of doing it - thanks for the suggestion !